crates/wasm/src/view_server.rs

```diff
@@ -338,13 +331,7 @@ impl ViewServer {
 
         self.latest_height = block.height;
 
-        let result = ScanBlockResult {
-            height: self.latest_height,
-            nct_updates: Default::default(),
-            new_notes,
-            new_swaps,
```

Comment:
When I later to `get_updates()` does it return every note stored on this struct despite the range I ask for? Does that mean when I do `get_updates()` many times, I keep getting that same list, but bigger? We may need to think through this more.

Reply:
Hmm, we can just clear notes and swaps during the `get_updates()` call

Reply:
Would clearing the notes affect this line?

```rust
          StatePayload::RolledUp(commitment) => {
                    if let std::collections::btree_map::Entry::Occupied(mut e) =
                        self.notes.entry(commitment)
                    {
```

I'm not sure what `RolledUp` is

Reply:
Also, if we move to this method (get_updates() clears state), we probably can start storing:
- last_position
- last_forgotten

On the struct itself. And when an get_updates() is called, we update those struct fields with the updates:
- set_position
- set_forgotten

Reply:
And likely we should change `get_updates` do something like `flush_updates`.

Reply:
```javascript
      StatePayload::RolledUp(commitment) => {
                    if let std::collections::btree_map::Entry::Occupied(mut e) =
                        self.notes.entry(commitment)
                    {
``` 
Now it seems to me that it's actually a mistake.
The `RolledUp` handler is associated with `swap claim`. When a swap occurs we save the advice so that in the future, when a `swap claim` occurs, we can use it to generate a `SpendableNote`
I'll test it

Reply:
```rust
 StatePayload::RolledUp(commitment) => {
                    if let std::collections::btree_map::Entry::Occupied(mut e) =
                        self.notes.entry(commitment)
                    {
``` 
This condition was indeed wrong. It is almost impossible to reproduce the bug related to it, because swap claim is automatically executed immediately after swap.

Reply:
I removed it, and tested correctly scanning the block with the swap claim.

Reply:
And after that, completed the redesign of `get_updates() `into `flush_updates()`